test/Sentry.Tests/SentrySdkTests.cs

```diff
@@ -270,6 +271,32 @@ public async Task Init_WithCache_BlocksUntilExistingCacheIsFlushed()
                 .Should().BeEmpty();
         }
 
+        [Fact]
+        public async Task Init_WithCustomTransport_UsesCaching()
+        {
+            // Arrange
+            using var cacheDirectory = new TempDirectory();
+
+            using var _ = SentrySdk.Init(o =>
+            {
+                o.Dsn = ValidDsnWithoutSecret;
+                o.DiagnosticLogger = _logger;
+                o.CacheDirectoryPath = cacheDirectory.Path;
+                o.Transport = new FakeFailingTransport();
+            });
+
+            // Act
+
+            SentrySdk.CaptureMessage("some message");
+            await SentrySdk.FlushAsync(TimeSpan.FromSeconds(5));
+
+            // Assert
+            Directory
+                .EnumerateFiles(cacheDirectory.Path, "*", SearchOption.AllDirectories)
+                .ToArray()
+                .Should().NotBeEmpty();
```

Comment:
These tests are failing, but if you merge in the latest main I think they should work